bezier-rs/lib/src/lib.rs

```diff
@@ -721,6 +699,141 @@ impl Bezier {
 		});
 		result
 	}
+
+	/// Approximate a bezier curve with circular arcs.
+	/// The algorithm can be customized using the [ArcsOptions] structure.
+	pub fn arcs(&self, arcs_options: ArcsOptions) -> Vec<CircleArc> {
+		let ArcsOptions { maximize_arcs, error, max_iterations } = arcs_options;
+		if maximize_arcs {
+			return self.arcs_helper(0., 1., error, max_iterations);
+		}
+		let mut extrema: Vec<f64> = self.local_extrema().into_iter().flatten().collect::<Vec<f64>>();
+		extrema.append(&mut vec![0., 1.]);
+		extrema.dedup();
+		extrema.sort_by(|ex1, ex2| ex1.partial_cmp(ex2).unwrap());
+
+		let mut t1 = extrema[0];
+		extrema
+			.iter()
+			.skip(1)
+			.flat_map(|&t2| {
+				let arcs = self.arcs_helper(t1, t2, error, max_iterations);
+				println!("t1, t2: {}, {}", t1, t2);
+				t1 = t2;
```

Comment:
One of the main reasons I did it this way was to avoid having to push to a vector, but to be fair I'm not sure if that makes that much of a difference in performance.

Reply:
Optimizing for performance is a valid enough reason, I say just leave it as is then.

Reply:
Another way to do this that avoids side effects is: 
```
extrema_t_list
  .iter()
  .take(extrema_t_list.len() - 1)
  .zip(extrema_t_list.iter().skip(1))
  .flat_map(|(&t1, &t2)| self.arcs_helper(t1, t2, error, max_iterations))
  .collect::<Vec<CircleArc>>()
```
Any thoughts on this?

Reply:
We found out about the `window` function and used that instead